student.cpp

```diff
@@ -243,10 +267,23 @@ float Student::GradeablePercent(GRADEABLE_ENUM g) const {
     if (my_max > 0) {
       sum += p * s / my_max;
     }
+    //   to add percentages only for non-extra credit gradeables
+    if (m != 0.0) {
+      sum_percentage += p;
+    }
+  }
+  assert(sum_percentage <= 1.0);
+  if (sum_max == 0) { // pure extra credit category
+    sum_percentage = 1.0;
   }
 
-  float percentage = GRADEABLES[g].hasSortedWeight() ? sum : GRADEABLES[g].getPercent() * sum;
+
+  //float percentage = GRADEABLES[g].hasSortedWeight() ? sum : GRADEABLES[g].getPercent() * sum;
+  float percentage = GRADEABLES[g].hasSortedWeight() ? sum : GRADEABLES[g].getPercent() * sum / sum_percentage;
```

Comment:
dividing by sum_percentage is a new change but is the correct way. Also it won't affect percentage if sum_percentage is 1.00